File: Data Structures & Algorithms/max-water-container/submission-1.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int left = 0;
        int right = heights.size() - 1;
        int maxWater = 0;
        while (left < right) {
            int width = right - left;
            int h = min(heights[left], heights[right]);
            int area = width * h;
            maxWater = max(maxWater, area);
            if (heights[left] < heights[right]) {
                left++;
            } else {
                right--;
            }
        }
        return maxWater;
    }
};

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        count = 0
        lcs = set()
        left = 0
        for right in range(len(s)):
            while s[right] in lcs:
                lcs.remove(s[left])
                left+=1
            lcs.add(s[right])
            count = max(count, len(lcs))
        
        return count